modules/twinklexfd.js

```diff
@@ -51,13 +51,12 @@ var utils = {
 
 	/**
 	 * Add namespace name to page title if not already given
-	 * CAUTION: namespace name won't be added if a namespace (*not* necessarily
-	 * the same as the one given) already is there in the title
+	 * If a namespace is already in the title, it will be stripped
 	 * @param {string} title
 	 * @param {number} namespaceNumber
 	 */
 	addNs: function(title, namespaceNumber) {
-		var title_obj = mw.Title.newFromUserInput(title, namespaceNumber);
+		var title_obj = mw.Title.newFromUserInput(utils.stripNs(title), namespaceNumber);
```

Comment:
The original was a bit intentional.

`addNs('foobar', 10)` --> `Template:Foobar`
`addNs('Module:foobar', 10)` -->`Module:Foobar`
(where 10 is template NS)

though addNs isn't actually being used for templates/modules (but i think this would be useful for twinkleimage?)

Reply:
Word.  That was what I was getting at in my original comment on this:

>Makes this less useful and is heavy-handed, but then it would do what it says on the tin. Only used for cfds right now, so it's shouldn't hurt.

Reply:
Removed it in the interest of going forward.